packages/smooth_app/lib/pages/user_management/sign_up_page.dart

```diff
@@ -0,0 +1,311 @@
+            ),
+            if (_foodProducer) ...<Widget>[
+              const SizedBox(height: space),
+              SmoothTextFormField(
+                type: TextFieldTypes.PLAIN_TEXT,
+                controller: _brandController,
+                hintText: 'Producer/brand',
+                prefixIcon: const Icon(Icons.person),
+                autofillHints: const <String>[AutofillHints.name],
+                validator: (String? value) {
+                  if (value == null || value.isEmpty) {
+                    return 'Please enter a producer or a brand name';
+                  }
+                  return null;
+                },
+              )
+            ],
+            const SizedBox(height: space),
+            ListTile(
+              leading: Checkbox(
+                value: _subscribe,
+                onChanged: (final bool? value) {
+                  if (value != null) {
+                    setState(() => _subscribe = value);
+                  }
+                },
+              ),
+              title: Text(
+                  'I\'d like to subscribe to the Open Food Facts newsletter (Note: I can unsubscribe from it at all time)'),
+            ),
+            const SizedBox(height: space),
+            ElevatedButton(
+              onPressed: () async => _signUp(),
+              child: Text(
+                'Sign Up',
+                style: theme.textTheme.bodyText2?.copyWith(
+                  fontSize: 18.0,
+                  color: theme.colorScheme.surface,
+                ),
+              ),
+              style: ButtonStyle(
+                minimumSize: MaterialStateProperty.all<Size>(
+                  Size(size.width * 0.5, theme.buttonTheme.height + 10),
+                ),
+                shape: MaterialStateProperty.all<RoundedRectangleBorder>(
+                  RoundedRectangleBorder(
+                    borderRadius: BorderRadius.circular(300.0),
+                  ),
+                ),
+              ),
+            ),
+          ],
+        ),
+      ),
+    );
+  }
+
+  Future<void> _signUp() async {
+    _disagreed = !_agree;
+    setState(() {});
+    if (!_formKey.currentState!.validate() || _disagreed) {
+      return;
+    }
+    _popEd = false;
+    await _openSigningUpDialog();
+    // TODO(monsieurtanuki): then what?
+  }
+
+  Future<Status?> _openSigningUpDialog() async => showDialog<Status>(
+        context: context,
+        builder: (BuildContext context) {
+          OpenFoodAPIClient.register(
+            user: User(
+              userId: _userController.text,
+              password: _password1Controller.text,
+            ),
+            name: _displayNameController.text,
+            email: _emailController.text,
+            newsletter: _subscribe,
+            orgName: _foodProducer ? _brandController.text : null,
+          ).then<void>(
+            (final Status status) => _popSigningUpDialog(status),
+          );
```

Comment:
We should directly log in the user here

Reply:
@M123-dev From what I read in `register`, it's probably already done:
```dart
   } else if (status.body!.contains('loggedin')) {
      return Status(status: 201);
 ```